### PR TITLE
feat(gatsby-remark-embed-snippet): apply gatsby-remark-prismjs configuration to embedded snippets

### PR DESCRIPTION
**Pull request by timhagn**
## Description

This is the continuation of PR #13841, which I sadly botched % ).
Added my changes here again on a clean fork.

While writing a documentation page for `gbi`, I noticed that line numbering and some other options of `gatsby-remark-prismjs` weren't working here. Stepping through it with Intellij Idea's Debugger confirmed that the [`visit(markdownAST, `code`, node => {...}`](https://github.com/gatsbyjs/gatsby/blob/f4e8d618e4cc77624df9b3d1779b2dcac2fdfcc5/packages/gatsby-remark-prismjs/src/index.js#L22) branch of it was of course never reached, as this plugin already had processed it to HTML. 
So I changed the constructed `node.type` to `code`, just gave the read in file as `node.value` and added `node.lang` to the returned node, which are needed for further processing down the line.
It just seemed unnecessary to duplicate `gatsby-remark-prismjs`'s code-highlighting functionality by creating _nearly_ the same output (and it should prevent any future deviations).

## Related Issues
Fixes the issue I would have opened otherwise ; ).

### CODE REVIEW

**Comment by pieh** [comment]
It looks good 👍 

I think this change warrant doing major version bump for `gatsby-remark-embed-snippet` as there is potential for changed behaviour when using it with same configuration (i.e added line numbering).

**Comment by timhagn** [comment]
Hi @pieh, thanks for the thumbs up : )!

Though as I already tested it in `reactjs.org` (see [this comment](https://github.com/gatsbyjs/gatsby/pull/13841#issuecomment-489408943) from the first PR #13841) and it doesn't seem to break anything in combination with the current `gatsby-remark-prismjs` - and `-embed-snippets` now just follows its default node creation / transformation process - I don't think a major version bump would really be necessary. 
But that's up to you people, of course ; ).

**Comment by pieh** [comment]
What I meant by changed behaviour, is that if you specify line numbering option (right now), it won't be used for embedded snippets. If I understand changes here correctly - this will fix it, so fixing this, also changes behaviour (different output for same plugin options) ;)

**Comment by timhagn** [comment]
\*hmmz\* But it fixes behavior in a backwards compatible manner, or doesn't it? Just thinking according to SemVer that'd be a minor version update. But what's semver more than "semantics" - so let's not dive into them ; ). In the end it's up to you and the other maintainers anyway ^^.

**Comment by timhagn** [comment]
Anyone willing to give this a review, please : )?

**Comment by timhagn** [comment]
@marcysutton @KyleAMathews @DSchau
Is this still wanted and would anyone of you be willing to give it a review, please : )?
What do you think about @pieh's objection? Would be willing to add an option for old vs. "new" behavior in the meantime... \*thinks\*

File: packages/gatsby-remark-embed-snippet/README.md [hunk 1]
```diff
@@ -164,7 +164,11 @@ function App() {
 
 ## How to use
 
-**Important**: This module must appear before `gatsby-remark-prismjs` in your plugins array, or the markup will have already been transformed into a code block and this plugin will fail to detect it and inline the file.
+**Important**: This module must appear before `gatsby-remark-prismjs` in your
+plugins array, or the markup will have already been transformed into a code
+block and this plugin will fail to detect it and inline the file.
+For further information about its options, visit `gatsby-remark-prismjs`'s
```
[the remark this answers is not in the document]
Comment (timhagn) [reply packages/gatsby-remark-embed-snippet/README.md]: As a German I'm always trying not to use "the" extensively ; ). In this case you might be right, it seems easier on the eyes. Commiting it.

**Comment by timhagn** [comment]
Thanks for reviewing it, @DSchau! 
As written, I've committed your suggested change to the README, hope it doesn't break the overall tests now - as you and the team seem to have made quite some progress while I re-opened this PR : ).
In case it would (at 15 successful ones now, fingers crossed ; ) and before I accidentally break the PR again, what's the preferred way of updating a PR to `gatsbyjs` from upstream? Merging changes into my local repo?
What do you think about me adding a switch in the meantime? And no prob about testing it with reactjs.org, was an interesting insight for me : ).
Thanks again!

**Comment by pieh** [comment]
> what's the preferred way of updating a PR to `gatsbyjs` from upstream?

We do squash commits from PR, so merging `gatsbyjs/gatsby` `master` branch into is ok. Rebasing would work too. You can chose whatever workflow you are comfortable with. I'll update branch and do some local testing and hopefully merge this today. 

I will do major version bump - this was never a blocker, just a note.

**Comment by pieh** [comment]
Published `gatsby-remark-embed-snippet@4.0.0`

Also added some BREAKING CHANGES note in changelog - https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-remark-embed-snippet/CHANGELOG.md#400-2019-06-14

**Comment by timhagn** [comment]
Thanks a lot @pieh, was worth the wait ; ). And to you, too, of course, @DSchau! Let's just say the whole Gatsby team : )!

**Pull request merged**